src/test/HostActivationTests/DotNetBuilder.cs

```diff
@@ -65,6 +65,67 @@ public DotNetBuilder AddMicrosoftNETCoreAppFramework(string version)
             return this;
         }
 
+        /// <summary>
+        /// Add a mock of the Microsoft.NETCore.App framework with the specified version
+        /// </summary>
+        /// <param name="version">Version to add</param>
+        /// <remarks>
+        /// Product runtime binaries are not added. All the added mock framework will contain is hostpolicy,
+        /// a mock version of coreclr, and a minimal Microsoft.NETCore.App.deps.json.
+        /// </remarks>
+        public DotNetBuilder AddMicrosoftNETCoreAppFrameworkMockCoreclr(string version)
+        {
+            // ./shared/Microsoft.NETCore.App/<version> - create a mock of the root framework
+            string netCoreAppPath = Path.Combine(_path, "shared", "Microsoft.NETCore.App", version);
+            Directory.CreateDirectory(netCoreAppPath);
+
+            // ./shared/Microsoft.NETCore.App/<version>/coreclr.dll - this is a mock, will not actually run CoreCLR
+            string hostPolicyFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("hostpolicy");
+            File.Copy(
+                Path.Combine(_repoDirectories.Artifacts, "corehost", hostPolicyFileName),
+                Path.Combine(netCoreAppPath, hostPolicyFileName),
+                true);
+
+            // ./shared/Microsoft.NETCore.App/<version>/coreclr.dll - this is a mock, will not actually run CoreCLR
+            string mockCoreclrFileName = RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("mockcoreclr");
+            File.Copy(
+                Path.Combine(_repoDirectories.Artifacts, "corehost_test", mockCoreclrFileName),
+                Path.Combine(netCoreAppPath, RuntimeInformationExtensions.GetSharedLibraryFileNameForCurrentPlatform("coreclr")),
+                true);
+
+            string netCoreAppPathDepsJson = Path.Combine(netCoreAppPath, "Microsoft.NETCore.App.deps.json");
+
+            string depsJsonBody = $@"{{
+              ""runtimeTarget"": "".NETCoreApp"",
+              ""targets"": {{
+                "".NETCoreApp"": {{
+                  ""Microsoft.NETCore.App/{version}"": {{
+                    ""native"": {{
+                      ""runtimes/any/native/libcoreclr.so"": {{ }}
+                    }}
+                  }},
+                  ""runtime.any.Microsoft.NETCore.DotNetHostPolicy/{version}"": {{
+                    ""native"": {{
+                      ""runtimes/any/native/libhostpolicy.so"": {{}}
+                    }}
+                  }}
+                }}
+              }},
+              ""libraries"": {{
+                ""Microsoft.NETCore.App/{version}"": {{
+                  ""type"": """",
+                  ""serviceable"": true,
+                  ""sha512"": """",
+                  ""path"": """",
+                  ""hashPath"": """"
```

Comment:
I think we can leave off `path` and `hashPath`

Reply:
Done.  I actually thought I tried that...